src/components/NewCommentForm/NewCommentForm.tsx

```diff
@@ -1,39 +1,97 @@
-import React from 'react';
+import React, { SyntheticEvent, useCallback, useState } from 'react';
+import { addPostCommentById } from '../../api/comments';
+import { Loader } from '../Loader';
 import './NewCommentForm.scss';
 
-export const NewCommentForm: React.FC = () => (
-  <form className="NewCommentForm">
-    <div className="form-field">
-      <input
-        type="text"
-        name="name"
-        placeholder="Your name"
-        className="NewCommentForm__input"
-      />
-    </div>
+type Props = {
+  selectedPostId: number;
+  getPostDetails: () => Promise<void>;
+};
 
-    <div className="form-field">
-      <input
-        type="text"
-        name="email"
-        placeholder="Your email"
-        className="NewCommentForm__input"
-      />
-    </div>
+export const NewCommentForm: React.FC<Props> = React.memo(({
+  selectedPostId,
+  getPostDetails,
+}) => {
+  const [name, setName] = useState('');
+  const [email, setEmail] = useState('');
+  const [body, setBody] = useState('');
+  const [isUploadComment, setIsUploadComment] = useState(false);
 
-    <div className="form-field">
-      <textarea
-        name="body"
-        placeholder="Type comment here"
-        className="NewCommentForm__input"
-      />
-    </div>
+  const catchSubmit = useCallback(async (e: SyntheticEvent) => {
+    e.preventDefault();
 
-    <button
-      type="submit"
-      className="NewCommentForm__submit-button button"
-    >
-      Add a comment
-    </button>
-  </form>
-);
+    setIsUploadComment(true);
+
+    const preparedObject = {
+      postId: selectedPostId,
+      name,
+      email,
+      body,
+    };
+
+    await addPostCommentById(preparedObject);
+    getPostDetails();
```

Comment:
addPostCommentById can return created comment
<img width="884" alt="Screenshot 2022-05-26 at 12 46 42" src="https://user-images.githubusercontent.com/47606379/170463467-50dad64b-74f9-4375-8693-dcdaf0db9370.png">
you don't need call getPostDetails again, you can add new comment to comments array